crates/wasm-mutate/src/lib.rs

```diff
@@ -145,6 +185,16 @@ impl WasmMutate {
         self
     }
 
+    pub(crate) fn consume_fuel(&mut self, qt: u64) -> bool {
```

Comment:
This should return `Result<()>` like `Resources::consume` does.

And it can take `&self` if the `fuel` member becomes a `std::cell::Cell`.